libflux/src/flux/semantic/flatbuffers/types.rs

```diff
@@ -189,8 +189,8 @@ impl From<fb::Prop<'_>> for Option<Property> {
 impl From<fb::Fun<'_>> for Option<Function> {
     fn from(t: fb::Fun) -> Option<Function> {
         let args = t.args()?;
-        let mut req = HashMap::new();
-        let mut opt = HashMap::new();
+        let mut req = BTreeMap::new();
+        let mut opt = BTreeMap::new();
```

Comment:
Curious about the choice here. Why switch to `BTreeMap`? I would think that for our purposes, neither data structure really provides much advantage over the other.

Reply:
If a type error is discovered in one of the arguments of a function, the error message will display the argument with the invalid type. In the case there are multiple invalid argument types, the type checker will return the first one it finds. This of course is dependent on the order in which we iterate over the arguments, which is nondeterministic in the case of a HashMap, deterministic in the case of a BTreeMap.

Changing `HashMap` references to `BTreeMap` was a quick fix to get tests passing. However I think it makes more sense to store argument types in a list in the order they're specified by the user. So I'll probably update this PR accordingly.